specification/cognitiveservices/data-plane/Language/preview/2022-10-01-preview/analyzetext.json

```diff
@@ -0,0 +1,3039 @@
+        },
+        "text": {
+          "type": "string",
+          "description": "The input text to process."
+        },
+        "language": {
+          "type": "string",
+          "description": "(Optional) This is the 2 letter ISO 639-1 representation of a language. For example, use \"en\" for English; \"es\" for Spanish etc. For Auto Language Detection, use \"auto\". If not set, use \"en\" for English as default."
+        }
+      }
+    },
+    "LanguageInput": {
+      "type": "object",
+      "required": [
+        "id",
+        "text"
+      ],
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Unique, non-empty document identifier."
+        },
+        "text": {
+          "type": "string"
+        },
+        "countryHint": {
+          "type": "string"
+        }
+      }
+    },
+    "AbstractiveSummarizationLROTask": {
+      "type": "object",
+      "description": "An object representing the task definition for an Abstractive Summarization task.",
+      "required": [
+        "parameters"
+      ],
+      "properties": {
+        "parameters": {
+          "$ref": "#/definitions/AbstractiveSummarizationTaskParameters"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/AnalyzeTextLROTask"
+        }
+      ],
+      "x-ms-discriminator-value": "AbstractiveSummarization"
+    },
+    "AbstractiveSummarizationTaskParameters": {
+      "type": "object",
+      "description": "Supported parameters for the pre-build Abstractive Summarization task.",
+      "allOf": [
+        {
+          "$ref": "common.json#/definitions/AbstractiveSummarizationTaskParametersBase"
+        },
+        {
+          "$ref": "common.json#/definitions/PreBuiltTaskParameters"
+        }
+      ]
+    },
+    "AbstractiveSummarizationLROResult": {
+      "type": "object",
+      "description": "An object representing the results for an Abstractive Summarization task.",
+      "properties": {
+        "results": {
+          "$ref": "#/definitions/AbstractiveSummarizationResult"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/AnalyzeTextLROResult"
+        }
+      ],
+      "required": [
+        "results"
+      ],
+      "x-ms-discriminator-value": "AbstractiveSummarizationLROResults"
+    },
+    "AbstractiveSummarizationResultBase": {
+      "type": "object",
+      "description": "An object representing the summarization results of each document.",
+      "properties": {
+        "documents": {
+          "type": "array",
+          "description": "Response by document",
+          "items": {
+            "allOf": [
+              {
+                "$ref": "#/definitions/AbstractiveSummaryDocumentResult"
+              },
+              {
+                "$ref": "#/definitions/DocumentDetectedLanguage"
+              }
+            ]
+          }
+        }
+      },
+      "required": [
+        "documents"
+      ]
+    },
```

Comment:
Question: I'm comparing the definition of "AbstractiveSummarizationResult" versus "ExtractiveSummarizationResult" and noticed that the former was split in two, where "AbstractiveSummarizationResultBase" is its own definition. Should it be merged back into one single definition to be consistent with "ExtractiveSummarizationResult", or is there a reason why they are separate?

Reply:
AbstractiveSummarizationResult inherit from AbstractiveSummarizationResultBase is because AbstractiveSummarizationResult is for pre-build skill. We also have custom skill CustomAbstractiveSummarizationResult also inherit from AbstractiveSummarizationResultBase, but the feature is delayed and removed from Oct release. We might add CustomAbstractiveSummarizationResult back in next API release. For ExtractiveSummarizationResult, so far we only have pre-build skill and we don't have plan to have custom skill like CustomExtractiveSummarizationResult yet. Good suggestion. We can make the data structure aligned with each other in the future.